### ISSUE TITLE
Spelunker HD [NPUB30152] Sound slowdown

### ISSUE THREAD

**Issue by Augusto7743** [RPCS3/rpcs3#4904]
Spelunker HD run graphically perfect, but in RPCS3 recent versions when the game begin in few seconds the music has a slowdown that not return to normal speed.
RPCS3 previous version the music had normal speed. Is how the SPU is paused or running some tasks in low speed.
Tried enabling and disabling any CPU and GPU settings in RPCS3 configuration has the same result.
Vulkan run more fast than OpenGL. Have games that OpenGL is more fast than Vulkan.
Spelunker HD run 60 FPS.
Thanks for read.

CPU AMD FX-6300
RAM 4 GB
GPU GeForce GT 640 latest drivers
Windows 10 Pro 64 RS3

Below is LOG and the RPCS3.log

RPCS3 v0.0.5-7104-a19113025 Alpha | HEAD
AMD FX(tm)-6300 Six-Core Processor  | 6 Threads | 3.98 GiB RAM | AVXx

[Spelunker HD [NPUB30152].log](https://github.com/RPCS3/rpcs3/files/2208865/Spelunker.HD.NPUB30152.log)

**Comment by benderscruffy** [comment]
still happens
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4111471/RPCS3.log.gz)

**Comment by Augusto7743** [comment]
The sound slowdown in music is an version before 06-2018.
If start an game and not move the player character the music will run in correct speed for more time,  but when moving the player character the music will slowdown perhaps because play the first sound fx.

**Comment by Augusto7743** [comment]
The issue happen in 0.0.5-6906 and not is possible test previous version before of 0.0.5-6906 because was all deleted.
The issue is before of 0.0.5-6906.
Spelunker HD use SPU to sound. The sound is slow few seconds when begin the gameplay.

**Comment by Augusto7743** [comment]
using debugger the problem is related with the threads
PPU[0x1000007] _cellsurMixerMain 
PPU[0x1000008] sound thread 
SPU[0x0000100] _synth2 Thread 

log using 0.0.18-12718
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/7147148/RPCS3.log)

**Comment by Augusto7743** [comment]
Not any try to fix ?
The game is with sound issue for 4 years.
The problem is RPCS3 change a thread priority to a lowest level to a thread creating the music.
Is possible see the priority being changed when using debugger.
If RPCS3 have a option in debugger to change thread priority perhaps is possible debug because the game have music issue.
Perhaps the fix is simple , but look how if the game not is priority to fix.
Bugged since 01-2018.

**Comment by MSuih** [comment]
Spelunker has been reported as playable, so it means that the reporter considers audio to be good enough.

Both systems here are below our recommended system specifications, and I suspect that it's the reason why audio doesn't sound as good. Audio is known to be among the first things to break when performance is not good enough, and it can happen even if framerate is still acceptable.

I'm closing this issue, it can be reopened if someone with a better system can confirm this. IF you're not sure what specs are recommended for decent performance in playable games, see our [quickstart guide](https://rpcs3.net/quickstart).

**Comment by Augusto7743** [comment]
Closed issue ?
Spelunker is playable , but since 01-2018 the music tempo was broken.
The used hardware (FX 6 cores) have power enough to play Spelunker.
In Linux is possible see cleary using htop RPCS3 creating few threads and using less of half of each core. Really only 3 cores are used.

The music tempo was broken before when SPU LLVM was added in RPCS3.
Any setting the music tempo has issues.
When the gameplay begin the music run in correct tempo , but in few seconds the music tempo change to slow.
Possibly was added an SPU performance code changing the thread priority how if the thread is having activity thus doing the issue.

AMD FX is decent to run several simple PSN games and even few discs games and Spelunker is in the list.
Closing the issue will have less change to an fix .... 4 years with sound broken ... have others users saying the same.
Please try play Spelunker and reply if is good sense say the sound is good enough. Not good gameplay even. The gameplay use music "alarms" when anything is coming and music tempo slow is hard understand.
Please reopen the issue.

**Comment by Yahfz** [comment]
![image](https://user-images.githubusercontent.com/17715080/147347172-0fa792df-b8ef-48a2-b7bf-2ae4adc7be0e.png)
just played for a few minutes and the audio is perfect here, how to reproduce then?

**Comment by Augusto7743** [comment]
The music tempo change when you pass the 2 fires.
If the player character stay where is being displayed in screenshot the music tempo will be changed only after of several minutes.
Move the player character to right ... after the 2 fires catch the first item and after the key above in left side ... the music tempo will change ..
If you have opened the debugger you see being changed a priority in a thread from normal speed to a low priority and if you click to stop the thread the music stop.
Possibly a SPU optimization code was done and the music was broken.
Spelunker use few CPU resources. Perhaps is possible run in a CPU 2 core.
Several earlier PSN games not use SPUs or use SPU only for sound (see interview with Insomniac games developer about PS3 Cell coding).

The game change music when the player catch some items and thus is possible understand when was stopped item power or what type of ghost comes in screen.
That game is hardest and the music tempo being slow the gameplay is much more hard yet.
Have moments you stop the gameplay.

**Comment by Augusto7743** [comment]
@MSuih
See the log attachment above of user @benderscruffy
The log detail how was tested using an Intel Core i7 CPU 920 @ 2.67GHz | 8 Threads | 7.99 GiB RAM | TSC: 2.66GHz and him confirming my report.
He have tested version 0.0.8-9436-e4ba0961. but the bug happen with current versions.
Spelunker is with sound issue for 4 years.

That issue report is correct and need be reopened for others users and devs reply.

**Comment by Asinin3** [comment]
i7 920 is far below recommended specs these days..

**Comment by Augusto7743** [comment]
Oh ...
Possibly Spelunker and TMNT Danger of Ooze are games requiring low cpu resources.
Totally sure both games run in an good dual core.
AMD FX 6 or 8 cores and i7 are enough to play Spelunker.
I understand "race condition" and some RPCS3 tasks need to be sync to run correctly , but that game use much less of 50 % my CPU and yet seeing in htop (task manager) the cores being used by RPCS3 not are at 100 %.
Spelunker music tempo was broken in 01-2018.
If music tempo is slow you not understand exaclty what detail is happening in game and after few minutes you close the game.
Not is possible play the game with music slow, you not understand when will coming the big ghost or items timing.

I not wait play Uncharted , Killzone or etc ...

Not have one dev replying about Spelunker and the issue is correct, but all time is reply invaliding not really testing and yet closed ...

Have a reply above saying
"just played for a few minutes and the audio is perfect here, how to reproduce then ?"
I had replied how to reproduce ...
Strangely the issue happen after the player goes to another side of screen and the screenshot is before in begin of stage.

If the game not is using core at 100 % a fix is possible and the log attachment is about a i7 much more faster than AMD FX and even thus happen the issue with music tempo.

All priority is in Dark Souls , GT , KZ and other games ?

**Comment by MSuih** [comment]
I tested this game for a bit on my ryzen 3700x, didn't hear any audio issues at all. Made it from start to 195m depth, so I think I went well past whatever point this was supposed to happen.

Remember that i7 is just a model range, it only tells how powerful the cpu is for that generation and isn't an indicator of how powerful it is in general. His i7 920 is from 2008 so it's 13 years old now and only two years newer than the console he's trying to emulate it with. Your FX-6300 is newer, but it's based on the Bulldozer architecture which was notorious for being underperforming and having "fake" cores thanks to it sharing some execution units between two cores.

Both your and his CPUs are well below our recommended system configurations. For AMD we recommend 6 cores and 12 threads or 8 cores from Zen 2 or newer, and for Intel we recommend same core configurations from Skylake or newer.

**Comment by Yahfz** [comment]
> Have a reply above saying "just played for a few minutes and the audio is perfect here, how to reproduce then ?" I had replied how to reproduce ... Strangely the issue happen after the player goes to another side of screen and the screenshot is before in begin of stage.

@Augusto7743 Look at the mini-map in the screenshot i posted. I clearly have explored the other side where you said the issue happens. The mini-map shows the area that was explored so it's pretty obvious that i've been to the other side. the screenshot shows me standing on the other side because i went back and forth to see if i could reproduce the issue, which i couldn't.

**Comment by Augusto7743** [comment]
@MSuih
Thanks very much for testing and reply me.

"Your FX-6300 is newer, but it's based on the Bulldozer architecture which was notorious for being underperforming and having "fake" cores thanks to it sharing some execution units between two cores."
You are correct.
FX-6300 and 8000 series have really 6 and 8 cores, but for each 2 cores is being shared 1 FPU unit thus comparing with a intel in same release date the FX are half perfomance.

@Yahfz
Thanks for your reply.
For me the screenshot had looked how if was to right and returned to left without catch items.

I have others games running very well , but Spelunker is the only with issue in music.
Not demanding fix , but reporting.
The last version version without issues with FX is between 12-2017 and 01-2018.
I remember the version running without issue was required one file in RPCS3 folder ... In moment not remember clearly ... ssl ? The requirement was removed in next versions.
Unhappily the download link was removed in history downloads.

If the game is started , but not moving the player character only after much time happen issues in music tempo.
Tested with rpcs3-v0.0.19-13137-cb2748ae_linux64 using default settings and cpu power profile to performance.
Using debugger the thread used to create music is "SPU[0x0000100] _synth2 Thread".
When happen the issue thread "SPU[0x0000100] _synth2 Thread" run more slow.
See the screenshots.

In gaming

![1](https://user-images.githubusercontent.com/21184987/147743062-ad82ce82-5294-4fef-8c12-3f486cbab8ec.png)


Htop taskmanager cpu usage

![2](https://user-images.githubusercontent.com/21184987/147743131-9f0fe5c7-b32f-43ad-ac08-f4e7172f3361.png)


Seeing in debugger not have much threads being used.
